monai/handlers/metrics_saver.py

```diff
@@ -0,0 +1,142 @@
+        self.batch_transform = batch_transform
+        self.compute_summary = compute_summary
+        self.save_rank = save_rank
+        self._filenames: List[str] = list()
+
+    def attach(self, engine: Engine) -> None:
+        """
+        Args:
+            engine: Ignite Engine, it can be a trainer, validator or evaluator.
+        """
+        engine.add_event_handler(Events.STARTED, self._started)
+        engine.add_event_handler(Events.ITERATION_COMPLETED, self._get_filenames)
+        engine.add_event_handler(Events.EPOCH_COMPLETED, self)
+
+    def _started(self, engine: Engine) -> None:
+        self._filenames = list()
+
+    def _get_filenames(self, engine: Engine) -> None:
+        if self.metric_details is not None:
+            self._filenames += self.batch_transform(engine.state.batch)["filename_or_obj"]
+
+    def __call__(self, engine: Engine) -> None:
+        """
+        Args:
+            engine: Ignite Engine, it can be a trainer, validator or evaluator.
+        """
+        if not os.path.exists(self.save_dir):
+            os.makedirs(self.save_dir)
+
+        ws = idist.get_world_size()
+        if self.save_rank >= ws:
+            raise ValueError("target rank is greater than the distributed group size.s")
+
+        if self.metrics is not None and len(engine.state.metrics) > 0:
+            if idist.get_rank() == self.save_rank:
+                # only save metrics to file in specified rank
+                with open(os.path.join(self.save_dir, "metrics.csv"), "w") as f:
+                    for k, v in engine.state.metrics.items():
+                        if k in self.metrics or "*" in self.metrics:
+                            f.write(f"{k}\t{str(v)}\n")
+
+        if (
+            self.metric_details is not None
+            and hasattr(engine.state, "metric_details")
+            and len(engine.state.metric_details) > 0
+        ):
+            _filenames = "\t".join(self._filenames)
+
+            if ws > 1:
+                if get_torch_version_tuple() > (1, 6, 0):
+                    # all gather across all processes
+                    _filenames = "\t".join(idist.all_gather(_filenames))
+                else:
+                    raise RuntimeError(
+                        "MetricsSaver can not save metric details in distributed mode with PyTorch < 1.7.0."
+                    )
+            if idist.get_rank() == self.save_rank:
+                _files = _filenames.split("\t")
+                for k, v in engine.state.metric_details.items():
+                    if k in self.metric_details or "*" in self.metric_details:
+                        v = v.cpu().numpy()
+                        v = np.concatenate([v, np.nanmean(v, axis=1, keepdims=True)], axis=1)
+                        with open(os.path.join(self.save_dir, k + "_raw.csv"), "w") as f:
+                            row_labels = "\t".join(["class" + str(i) for i in range(v.shape[1] - 1)]) + "\taverage"
```

Comment:
`\t` could be a user-specified delimiter defaults to `\t`

Reply:
OK, added `delimiter` arg in the latest commit.
Thanks.